API_Movie_Call.py

```diff
@@ -12,31 +12,103 @@
 
 # Hard coded year month and day will be replaced once we are able to pass
 # a date object that these variables can be extrated from. 
+# Day month year need to be in string format because the API is expecteing a ####-##-## format for Year-Month-Day
 ##################
-year = '2008'
-month = '10'
-day = '22'
+year = '2010'
+month = '08'
+day = '23'
 ##################
 
 # Gets the API key from my local system variables.
 key = os.environ.get('MOVIE_KEY')
 
 # Base URL for discovering movies.
-movie_url = 'https://api.themoviedb.org/3/discover/movie?'
+movie_discover_url = 'https://api.themoviedb.org/3/discover/movie?'
+# URL for the images, w500 is not part of the base url, but it is the max width for a poster.
+movie_images_url = 'https://image.tmdb.org/t/p/w500/'
 
-# Query for discovery
-query = {'api_key':key, 
+def main():
+    query =set_api_discover_params(year, month, day, key)
+    data = api_movie_request(movie_discover_url, query)
+    movie_raw = get_movie_results(data)
+    movie_title, movie_overview, movie_poster_path = get_movie_details(movie_raw)
+    movie_img_link = get_movie_image(movie_poster_path)
+
+    print(movie_title)
+    print(movie_overview)
+    print(movie_img_link)
+
+def set_api_discover_params(year, month, day, key) -> dict:
+    '''
+    Takes a date object seperated by components and an API key and returns the 
+    api call query parameters.
+    '''
+    # TODO potential if statement to check year, month, day are type string.
+    # Query for discovery
+    query = {'api_key':key, 
     'language':'en-US', # Language sets the movies title string so we return english version names of movies
     'sort_by':'release_date.desc', # Sorts the results of the API Call.
     'include_adult':'false',
+    'include_video':'false',
     'page':'1', # Page represents what page of the results will be looked at.
     'primary_release_year':year, 
     'primary_release_date.lte':f'{year}-{month}-{day}',
     'vote_average.gte':'3', 
     'with_original_language':'en'}
 
-data = requests.get(movie_url, params=query).json()
-movie_results_list = data['results']
+    return query
+
+def api_movie_request(url, query):
+    '''
+    Takes a url and a query string and returns json data.
+    '''
+
+    data = requests.get(url, params=query).json()
+
+    return data
+
+def get_movie_results(json_data) -> dict:
+    '''
+    Takes json data returned from an api request and returns one movie
+    (Break down into two functions? Might make unit testing easier.)
+    '''
+    movie_result_list = json_data['results']
+    if len(movie_result_list) > 0:
+        return movie_result_list[0]
+    else:
+        return 'Your movie is in another castle.'
```

Comment:
Possibly want to return None or something that is really, definitely, not a movie?